### PR TITLE
Allow custom password length.

### PR DESCRIPTION
**Pull request by jmcarp**
Fixes https://github.com/cloudfoundry/bosh-cli/issues/302.

cc @drnic @Rukenshia

### CODE REVIEW

File: types/certificate_generator.go [hunk 1]
```diff
@@ -194,7 +194,7 @@ func stringInArray(key string, list []string) bool {
 	return false
 }
 
-func objToStruct(input interface{}, str interface{}) error {
+func objToStruct(input interface{}, str interface{}, supportedParameters []string) error {
```
Comment (dpb587-pivotal) [inline types/certificate_generator.go]: Think we should move this to a separate file now that it's shared between types?

File: types/password_generator.go [hunk 2]
```diff
@@ -3,23 +3,40 @@ package types
...
+
 func NewPasswordGenerator() ValueGenerator {
 	return passwordGenerator{}
 }
 
 func (passwordGenerator) Generate(parameters interface{}) (interface{}, error) {
+	var params passwordParams
+	err := objToStruct(parameters, &params, supportedPasswordParams)
+	if err != nil {
+		return nil, errors.WrapError(err, "Failed to generate password, parameters are invalid")
```
Comment (dpb587-pivotal) [inline types/password_generator.go:32]: We should probably have a test for this new code.

File: types/password_generator.go [hunk 3]
```diff
@@ -3,23 +3,40 @@ package types
...
+	}
+	if params.Length == 0 {
+		params.Length = DefaultPasswordLength
+	}
 
 	lengthLetterRunes := big.NewInt(int64(len(letterRunes)))
-	passwordRunes := make([]rune, DefaultPasswordLength)
```
Comment (dpb587-pivotal) [inline types/password_generator.go:22]: If a default is configured... but it can't be changed... is it really a default? hehe

**Comment by cppforlife** [comment]
added https://www.pivotaltracker.com/story/show/157680118 to the queue

**Comment by jmcarp** [comment]
Thanks @dpb587-pivotal ! Just pushed up some revisions.

**Comment by jmcarp** [comment]
Bump @cppforlife @dpb587-pivotal

**Comment by h4xnoodle** [comment]
@jmcarp did your revisions from @dpb587-pivotal 's comments on May 17 make it to github on May 18?

**Comment by jmcarp** [comment]
@h4xnoodle: yep, I added a test for unsupported parameters in https://github.com/cloudfoundry/config-server/pull/8/commits/b3bdd17d2a5f0afb011f84e69b3b645cc2ca2b3c.

**Comment by pivotal-jamil-shamy** [comment]
Should be included in the next BOSH CLI release

**Pull request merged**